### PR TITLE
Add prior_box output order control

### PR DESCRIPTION
**Pull request by baiyfbupt**
Fix https://github.com/PaddlePaddle/Paddle/issues/11734

### CODE REVIEW

File: python/paddle/fluid/layers/detection.py [hunk 1]
```diff
@@ -658,6 +658,7 @@ def prior_box(input,
               variance=[0.1, 0.1, 0.2, 0.2],
               flip=False,
               clip=False,
+              def_ord=True,
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: 考虑API的兼容性，新增的参数放到最后吧：

```python
steps = [0.0, 0.0],
offset=[0.5],
name=None,
min_max_aspect_ratios_order=False);
```

File: python/paddle/fluid/layers/detection.py [hunk 2]
```diff
@@ -683,6 +684,10 @@ def prior_box(input,
             Default:[0.1, 0.1, 0.2, 0.2].
        flip(bool): Whether to flip aspect ratios. Default:False.
        clip(bool): Whether to clip out-of-boundary boxes. Default: False.
+       def_ord(bool): Whether to output prior box in default order. If set True,
+            this layer will output in the order of [min, various_ar, max]
+            otherwise, this layer will output [min, max, various_ar]
+            Default: True.
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: Maybe  def_ord  ->  min_max_aspect_ratios_order  ？

If set True,  the output prior box is in order of [min, max, aspect_ratios], which is consistent with Caffe. Please note,  this order affects the weights order of convolution layer followed by and does not affect the fininal detection results.

Please also update the comments in C++ doc.

Comment (qingqing01) [reply python/paddle/fluid/layers/detection.py]: Please also update the argument of 

https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/layers/detection.py#L766

**Comment by qingqing01** [comment]
@reyoung @panyx0718 This PR will change the Python API, please help to review, Thanks!

File: python/paddle/fluid/layers/detection.py (hunk 1 above)
Comment (baiyfbupt) [reply python/paddle/fluid/layers/detection.py]: Done

File: python/paddle/fluid/layers/detection.py (hunk 2 above)
Comment (baiyfbupt) [reply python/paddle/fluid/layers/detection.py]: Done

File: paddle/fluid/operators/detection/prior_box_op.cc [hunk 3]
```diff
@@ -149,6 +149,13 @@ class PriorBoxOpMaker : public framework::OpProtoAndCheckerMaker {
                    "(float) "
                    "Prior boxes center offset.")
         .SetDefault(0.5);
+    AddAttr<bool>(
+        "min_max_aspect_ratios_order",
+        "(bool) If set True, the output prior box is in order of"
+        "[min, max, aspect_ratios], which is consistent with Caffe."
```
Comment (panyx0718) [inline paddle/fluid/operators/detection/prior_box_op.cc:155]: what if it's False?

**Review by panyx0718 — DISMISSED** [review]
Approve for the API change only. Please check other parts

File: paddle/fluid/operators/detection/prior_box_op.cc (hunk 3 above)
Comment (baiyfbupt) [reply paddle/fluid/operators/detection/prior_box_op.cc:155]: If it's False, this layer will output in the original order: [min, aspect_ratios, max]

**Review by qingqing01 — APPROVED** [review]

**Pull request merged**